plugins/modules/database.ps1

```diff
@@ -0,0 +1,244 @@
+#!powershell
+# -*- coding: utf-8 -*-
+
+# (c) 2022, John McCall (@lowlydba)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+#AnsibleRequires -PowerShell ansible_collections.lowlydba.sqlserver.plugins.module_utils._SqlServerUtils
+
+Import-ModuleDependency
+$ErrorActionPreference = "Stop"
+
+$spec = @{
+    supports_check_mode = $true
+    options = @{
+        sql_instance = @{type = 'str'; required = $true }
+        sql_username = @{type = 'str'; required = $false }
+        sql_password = @{type = 'str'; required = $false; no_log = $true }
+        database = @{type = 'str'; required = $true }
+        recovery_model = @{type = 'str'; required = $false; choices = @('Full', 'Simple', 'BulkLogged') }
+        data_file_path = @{type = 'str'; required = $false }
+        log_file_path = @{type = 'str'; required = $false }
+        owner_name = @{type = 'str'; required = $false; }
+        maxdop = @{type = 'int'; required = $false; }
+        secondary_maxdop = @{type = 'int'; required = $false; }
+        compatibility = @{type = 'str'; required = $false; }
+        rcsi = @{type = 'bool'; required = $false; }
+        state = @{type = 'str'; required = $false; default = 'present'; choices = @('present', 'absent') }
+    }
```

Comment:
```suggestion
    }
    required_together = @(
        ,@('sql_username', 'sql_password')
    )
```
These two are required together, no?
Details in : https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general_windows.html